### PR TITLE
nvme: return 1 when the device is not supported in nvme_probe_one

### PR DESCRIPTION
**Pull request by uchida2016**
When we try to add some nvme bdevs using rpc.py script with construct_nvme_bdev option, the first device is added successfully. But the second or later devices are failed to add. This patch fixes the issue.

### CODE REVIEW

File: lib/nvme/nvme.c [hunk 1]
```diff
@@ -248,6 +248,9 @@ nvme_probe_one(enum spdk_nvme_transport transport, spdk_nvme_probe_cb probe_cb,
 		ctrlr->probe_info = *probe_info;
 
 		TAILQ_INSERT_TAIL(&g_spdk_nvme_driver->init_ctrlrs, ctrlr, tailq);
+	} else {
+		/* The device is not supported */
+		return 1;
```
Comment (dverkamp-intel) [inline lib/nvme/nvme.c]: This part looks correct (the code has moved to nvme_ctrlr.c, but the logic should be the same).

File: lib/bdev/nvme/blockdev_nvme.c [hunk 2]
```diff
@@ -390,6 +390,10 @@ probe_cb(void *cb_ctx, const struct spdk_nvme_probe_info *probe_info,
 		return false;
 	}
 
+	if (ctx->controllers_remaining > 0) {
```
Comment (dverkamp-intel) [inline lib/bdev/nvme/blockdev_nvme.c]: I don't think the controllers_remaining change is necessary, and I think actually the controllers_remaining logic can be deleted entirely - can you skip this part of the patch and verify that it still works correctly?

Comment (uchida2016) [inline lib/bdev/nvme/blockdev_nvme.c]: Actually, I planed to submit another patch to delete controller_remaining and to change condition of for loop from "i < NVME_MAX_CONTROLLERS" to "i < ctx->num_whitelist_controllers" after this patch is merged. May I  put the two modification into this patch?

File: lib/nvme/nvme.c (hunk 1 above)
Comment (uchida2016) [inline lib/nvme/nvme.c]: nvme_probe_one is in nvme.c of the current GitHub code. Should I wait until you move it to nvme_ctrlr.c?

Comment (dverkamp-intel) [inline lib/nvme/nvme.c]: I merged this part of the fix (you are right, it is still in nvme.c, my mistake) as ea3a2772ac78f3646d0c98e9c7995bcf5ed1ee81 - thanks.

Comment (uchida2016) [inline lib/nvme/nvme.c]: Thanks.

File: lib/bdev/nvme/blockdev_nvme.c (hunk 2 above)
Comment (uchida2016) [inline lib/bdev/nvme/blockdev_nvme.c]: I'll submit this part as a new PR.

Comment (uchida2016) [inline lib/bdev/nvme/blockdev_nvme.c]: Let me confirm. Can we configure iscsi.conf as follows? If yes, the controllers_remaining change is necessary to prevent creating bdevs for 4 NVMe devices. If no, controllers_remaining can be deleted from nvme_probe_ctx structure.

[Nvme]
  BDF 0000:02:00.0
  BDF 0000:03:00.0
  BDF 0000:04:00.0
  BDF 0000:05:00.0
  NumControllers 2

Comment (dverkamp-intel) [inline lib/bdev/nvme/blockdev_nvme.c]: I'm not really sure what the behavior should be when NumControllers is set and explicit BDF addresses are also provided.  NumControllers is really mainly meant for use with the default setup where no BDFs are specified.  I think your interpretation is probably reasonable, or we could change the configuration file parsing code to disallow this case (e.g. only allow BDF or NumControllers, not both).

Comment (dverkamp-intel) [inline lib/bdev/nvme/blockdev_nvme.c]: Actually, re-reading the conf parsing code, it looks like NumControllers is indeed intended to be used with BDF, although I'm not really sure why (I'll check with some folks to see if this is really needed or if a plain list of BDFs would be good enough).

**Comment by dverkamp-intel** [comment]
I think this change has been superseded by other modifications to the [Nvme] config section - can you verify whether this pull request is no longer necessary?

**Comment by uchida2016** [comment]
This is no longer necessary. I close the PR.
